config/routes.rb

```diff
@@ -28,8 +28,11 @@
 
     resources :learning_units, only: %i[show] do
       get 'completed'
+      resources :resources, only: %i[index]
     end
 
+    resources :resources, only: %i[show]
```

Comment:
En consecuencia del comentario anterior, faltaría definir la ruta
Path: /resources/:id/average_evaluation
Controller#Action:  resources#average

creo que anidando un get 'average_evaluation', to: 'resources#average' adentro de esta fila (con un "do" y posteriormente un "end") funcionaría. Se puede probar desde la consola de rails con "rails routes".